cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -0,0 +1,376 @@
+		ResourceModel:   config,
+	}, nil
+}
+func ReadConfig(client mongodbatlas.Client, currentModel *Model) (*Model, handler.ProgressEvent, error) {
+	setup() // logger setup
+	projectID := *currentModel.ProjectId
+	ClusterName := *currentModel.ClusterName
+
+	// API call to read configuration
+	globalCluster, resp, err := client.GlobalClusters.Get(context.Background(), projectID, ClusterName)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			return nil, handler.ProgressEvent{
+				OperationStatus:  handler.Failed,
+				Message:          err.Error(),
+				HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+		}
+		return nil, handler.ProgressEvent{}, err
+	}
+	nameSpaces := globalCluster.ManagedNamespaces
+	zoneMappings := globalCluster.CustomZoneMapping
+
+	if len(nameSpaces) == 0 && len(zoneMappings) == 0 {
+		return nil, handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "resource Not Found",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+
+	readModel := newModel(globalCluster, currentModel)
+	_, _ = logger.Debugf("response Object:%+v", readModel)
+
+	return readModel, handler.ProgressEvent{}, nil
+}
+
+func newModel(globalCluster *mongodbatlas.GlobalCluster, currentModel *Model) *Model {
+	setup() // logger setup
+	readModel := new(Model)
+	readModel.ProjectId = currentModel.ProjectId
+	readModel.ClusterName = currentModel.ClusterName
+	maps := flattenManagedNamespaces(globalCluster.ManagedNamespaces)
+	zones := customZoneToModelMappings(globalCluster.CustomZoneMapping)
+	readModel.CustomZoneMappings = zones
+	readModel.ManagedNamespaces = maps
+	readModel.ApiKeys = currentModel.ApiKeys
+	readModel.RemoveAllZoneMapping = currentModel.RemoveAllZoneMapping
+	return readModel
+}
+
+func flattenManagedNamespaces(managedNamespaces []mongodbatlas.ManagedNamespace) []ManagedNamespace {
+	var results []ManagedNamespace
+
+	if len(managedNamespaces) > 0 {
+		for ind := range managedNamespaces {
+			namespace := ManagedNamespace{
+				Db:                     &managedNamespaces[ind].Db,
+				Collection:             &managedNamespaces[ind].Collection,
+				CustomShardKey:         &managedNamespaces[ind].CustomShardKey,
+				IsCustomShardKeyHashed: managedNamespaces[ind].IsCustomShardKeyHashed,
+				IsShardKeyUnique:       managedNamespaces[ind].IsShardKeyUnique,
+			}
+			results = append(results, namespace)
+		}
+	}
+	return results
+}
+
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	// No OP
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Update Complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	// No OP
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "List Complete",
+	}, nil
+}
```

Comment:
we shouldt be returning a success if the function is not imeplemented
```suggestion
func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
	return handler.ProgressEvent{}, errors.New("not implemented: Update")
}
func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
	return handler.ProgressEvent{}, errors.New("not implemented: List")
}
```